server/tables/tables.go

```diff
@@ -834,6 +834,26 @@ func PreAutoMigrate(db *gorm.DB) ([]PostAutoMigrateLogic, error) {
 			return nil
 		})
 	}
+
+	if m.HasTable("APIKeys") {
+		// Add nonce column w/o unique index so that we can backfill it before
+		// gorm adds the unique index.
+		if !m.HasColumn(&APIKey{}, "nonce") {
+			stmt := `ALTER TABLE "APIKeys" ADD COLUMN "nonce" varchar(255) DEFAULT NULL`
+			if err := db.Exec(stmt).Error; err != nil {
+				return nil, err
+			}
+		}
+		backfill := `
+			UPDATE "APIKeys"
+			SET nonce = SUBSTR(value, 1, 6)
```

Comment:
Is the idea that we're filling 'nonce' with arbitrary unique data that will later get overwritten by backfillUnencryptedKeys()?

Reply:
We use the first 6 characters of the key as the nonce, the only difference is that we were previously only populating this column when encryption was enabled. I've also removed the redundant update in backfillUnencryptedKeys.